src/System.AspNetCore/LogicalCallContext.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.Omex.System.Diagnostics;
+
+namespace Microsoft.Omex.System.AspNetCore
+{
+	/// <summary>
+	/// Context data stored on LogicalCallContext
+	/// </summary>
+	public class LogicalCallContext : LogicalCallContextBase
```

Comment:
It would be nice (but not mandatory) to merge LogicalCallContext and LogicalCallContextBase into a single class.